cmds/bmp/create_baremetals_command.go

```diff
@@ -45,10 +52,57 @@ func (cmd createBaremetalsCommand) Options() cmds.Options {
 
 func (cmd createBaremetalsCommand) Validate() (bool, error) {
 	cmd.printer.Printf("Validating %s command: options: %#v", cmd.Name(), cmd.options)
+	if cmd.options.Deployment == "" {
+		return false, errors.New("please specify the deployment file with -d")
+	}
+
+	_, err := os.Stat(cmd.options.Deployment)
+	if os.IsNotExist(err) {
+		return false, errors.New(fmt.Sprintf("deployment file %s doesn't exist", cmd.options.Deployment))
+	}
+
 	return true, nil
 }
 
 func (cmd createBaremetalsCommand) Execute(args []string) (int, error) {
 	cmd.printer.Printf("Executing %s comamnd: args: %#v, options: %#v", cmd.Name(), args, cmd.options)
+
+	filename, _ := filepath.Abs(cmd.options.Deployment)
+	yamlFile, err := ioutil.ReadFile(filename)
+
+	if err != nil {
+		errorMessage := fmt.Sprintf("bmp: could not read File %s, error message %s", filename, err.Error())
+		return 1, errors.New(errorMessage)
+	}
+
+	var deploymentInfo clients.Deployment
+	err = yaml.Unmarshal(yamlFile, &deploymentInfo)
+	if err != nil {
+		errorMessage := fmt.Sprintf("bmp: failed to decode Yaml File %s, error message %s", filename,
+			err.Error())
+		return 1, errors.New(errorMessage)
+	}
+
+	createBaremetalsInfo := clients.CreateBaremetalsInfo{}
+	createBaremetalsInfo.Deployment = deploymentInfo.Name
+	for _, resource := range deploymentInfo.ResourcePools {
+		if resource.CloudProperty.Baremetal {
+			createBaremetalsInfo.BaremetalSpecs = append(createBaremetalsInfo.BaremetalSpecs,
+				resource.CloudProperty)
+		}
+	}
+
+	createBaremetalsResponse, err := cmd.bmpClient.CreateBaremetals(createBaremetalsInfo, cmd.options.DryRun)
+	if err != nil {
+		return 1, err
+	}
+
+	if createBaremetalsResponse.Status != 200 {
```

Comment:
I am wondering if you should use the `common.IsHttpError(int)` function instead of looking for 200 here. At least motivate why 200 is the only valid response. Sometimes 201, 202, and other codes are valid. Make sure to check with @JimmyMa on this and what the server will respond here.